src/System.Drawing.Common/src/System/Drawing/Printing/PrintingServices.Unix.cs

```diff
@@ -587,9 +608,11 @@ private static PrinterResolution ParseResolution(string resolution)
 
         /// <summary>
         /// </summary>
-        private static void LoadPrinters()
+        private static Hashtable LoadPrinters(Hashtable printers)
         {
-            installed_printers.Clear();
+            Volatile.Write(ref installed_printers, null);
```

Comment:
What's the purpose of this here?  This will effectively become a nop once the printers have been initialized, because LoadPrinters won't be called again.

In what situation / when do we need to reset and re-load?  This write should be moved to that location.

Reply:
In the original behavior we were calling LoadPrinters each time we requested the InstalledPrinters. I think this behavior is desired, otherwise how do we force the list of printers to update when CUPS does?

Reply:
If the goal is to always have an up-to-date view, then why is there any caching at all?

Reply:
I was looking to resolve the multi-threading issue more than caching. I'm unsure of how to use the caching, as in theory at any time the system could add a new printer to CUPS. The caching between threads makes the most sense, and EnsureInitialized seems to do that. If I pass a lock object, I could have only one thread actually working instead of using the race condition. 

We also request the default printer, and I don't see a reason to re-run LoadPrinter there. But that was already taken care of with the PopulatedPrinters flag.

Reply:
I think what @stephentoub meant here is why do we want to cache the value and not every time populate/update the printers as it seems like we always want/need to have an up-to-date view. @stephentoub correct me if I'm wrong.

Reply:
Yeah, that makes sense to me. I was looking to use a single racing thread more than implement caching.

It would appear if I pass EnsureInitialized a sync object it will lock everything until the first thread finishes, then return a value. Otherwise, I can not pass an object and it will take the first thread that finishes.

I was thinking it's what you meant for it to be used for, more than the caching side from your comments earlier. 

> an optimistic concurrency scheme and allow multiple threads to race to build up results of which only one is published, I don't feel strongly about. The latter might be cleaner and less risky, though. You might be able to use LazyInitializer.EnsureInitialized to do it fairly cleanly.